packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -528,18 +663,153 @@ abstract class SliderComponentShape {
   /// If [labelPainter] is non-null, then [labelPainter.paint] should be
   /// called with the location that the label should appear. If the labelPainter
   /// passed is null, then no label was supplied to the [Slider].
+  void paint(
+      PaintingContext context,
+      Offset center, {
+        Animation<double> activationAnimation,
+        Animation<double> enableAnimation,
+        bool isDiscrete,
+        TextPainter labelPainter,
+        RenderBox parentBox,
+        SliderThemeData sliderTheme,
+        TextDirection textDirection,
+        double value,
+      });
+}
+
+// The following shapes are the material defaults.
+
+/// This is the default shape to a [Slider]'s track if no other shape is
+/// specified. It paints a solid colored rectangle, vertically centered in the
+/// parent box. The color is determined by the [Slider]'s enabled state and the
+/// track piece's active state. These colors are defined in the
+/// [SliderThemeData].
+///
+/// See also:
+///
+///  * [Slider] for the component that this is meant to display this shape.
+///  * [SliderThemeData] where an instance of this class is set to inform the
```

Comment:
[SliderTheme], which can be used to configure the the track shape of all sliders in a widget subtree.

Reply:
done